heartbeat/beater/heartbeat.go

```diff
@@ -211,20 +211,34 @@ func (bt *Heartbeat) Run(b *beat.Beat) error {
 
 	defer bt.scheduler.Stop()
 
-	<-bt.done
+	// Wait until run_once ends or bt is being shut down
+	waitMonitors.AddChan(bt.done)
```

Comment:
I don't remember if we ever waited for Kill signal from the OS or any interrupts? Not blocking the PR, just want to understand if we are handling anything differently here.

Reply:
Yes, any interrupt is handled by libbeat which in turn triggers `heartbeat.Stop()` and finally `<-bt.done`. This is just replicating the same condition using the wait signal approach and listening both for interrupts OR run once finished event, if in run once mode.